src/storage/src/boundary/tcp_boundary.rs

```diff
@@ -336,28 +338,29 @@ pub mod server {
         /// Construct a new pusher. It'll retain a weak reference to `client_token`.
         fn new(
             sender: UnboundedSender<R>,
-            token: Rc<dyn Any>,
+            token: Rc<RefCell<Option<Rc<dyn Any>>>>,
             client_token: &Arc<()>,
             convert: F,
         ) -> Self {
             Self {
                 sender,
                 client_token: Arc::downgrade(client_token),
                 convert,
-                token: Some(token),
+                token,
             }
         }
     }
 
     impl<T, D, R, F: Fn(EventCore<T, D>) -> R> EventPusherCore<T, D> for UnboundedEventPusher<R, F> {
         fn push(&mut self, event: EventCore<T, D>) {
+            let mut token = self.token.borrow_mut();
```

Comment:
is there any chance this panics? can both pushers be called at the same time?

Reply:
Nope, they run on the same thread!